zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/BackgroundTaskManagerFactoryTest.java

```diff
@@ -0,0 +1,234 @@
+        .anyMatch(task -> isProcessInstanceArchiverTask(task));
+  }
+
+  @Test
+  void shouldNotScheduleProcessInstanceArchiverTaskWhenConfigDisabled() {
+    // given
+    config.getHistory().setProcessInstanceEnabled(false);
+
+    // when
+    final var taskManager = factory.build();
+
+    // then
+    final var tasks = getTasksFromManager(taskManager);
+    assertThat(tasks)
+        .as("Should not contain ProcessInstancesArchiverJob when config is disabled")
+        .noneMatch(task -> isProcessInstanceArchiverTask(task));
+  }
+
+  @Test
+  void shouldScheduleProcessInstanceCountJobWhenConfigEnabledAndMetricsTrackingEnabled() {
+    // given
+    config.getHistory().setProcessInstanceEnabled(true);
+    config.getHistory().setTrackArchivalMetricsForProcessInstance(true);
+
+    // when
+    final var taskManager = factory.build();
+
+    // then
+    final var tasks = getTasksFromManager(taskManager);
+    assertThat(tasks)
+        .as("Should contain ProcessInstanceToBeArchivedCountJob when both configs are enabled")
+        .anyMatch(task -> isProcessInstanceToBeArchivedCountTask(task));
+  }
+
+  @Test
+  void shouldNotScheduleProcessInstanceCountJobWhenProcessInstanceConfigDisabled() {
+    // given
+    config.getHistory().setProcessInstanceEnabled(false);
+    config.getHistory().setTrackArchivalMetricsForProcessInstance(true);
+
+    // when
+    final var taskManager = factory.build();
+
+    // then
+    final var tasks = getTasksFromManager(taskManager);
+    assertThat(tasks)
+        .as("Should not contain ProcessInstanceToBeArchivedCountJob when PI config is disabled")
+        .noneMatch(task -> isProcessInstanceToBeArchivedCountTask(task));
+  }
+
+  @Test
+  void shouldNotScheduleProcessInstanceCountJobWhenMetricsTrackingDisabled() {
+    // given
+    config.getHistory().setProcessInstanceEnabled(true);
+    config.getHistory().setTrackArchivalMetricsForProcessInstance(false);
+
+    // when
+    final var taskManager = factory.build();
+
+    // then
+    final var tasks = getTasksFromManager(taskManager);
+    assertThat(tasks)
+        .as(
+            "Should not contain ProcessInstanceToBeArchivedCountJob when metrics tracking is disabled")
+        .noneMatch(task -> isProcessInstanceToBeArchivedCountTask(task));
+  }
+
+  @Test
+  void shouldScheduleBothProcessInstanceTasksWhenAllConfigsEnabled() {
+    // given
+    config.getHistory().setProcessInstanceEnabled(true);
+    config.getHistory().setTrackArchivalMetricsForProcessInstance(true);
+
+    // when
+    final var taskManager = factory.build();
+
+    // then
+    final var tasks = getTasksFromManager(taskManager);
+    assertThat(tasks)
+        .as("Should contain both PI archiver and count tasks when all configs are enabled")
+        .anyMatch(task -> isProcessInstanceArchiverTask(task))
+        .anyMatch(task -> isProcessInstanceToBeArchivedCountTask(task));
+  }
+
+  @Test
+  void shouldNotScheduleAnyProcessInstanceTasksWhenConfigDisabled() {
```

Comment:
is this not essentially the same as `shouldNotScheduleProcessInstanceCountJobWhenProcessInstanceConfigDisabled`? Not a big issue, but one test while asserting on both conditions is probably sufficient no?